components/OlasTokenPage/SupplyPieChart.jsx

```diff
@@ -0,0 +1,200 @@
+import PropTypes from 'prop-types';
+import React, { useState, useEffect } from 'react';
+import Web3 from 'web3';
+import { Pie } from 'react-chartjs-2';
+import { Chart, ArcElement } from 'chart.js';
+import Verify from '../Verify';
+import olasAbi from '../../data/ABIs/Olas.json';
+
+// manually register arc element – required due to chart.js tree shaking
+Chart.register(ArcElement);
+
+const olasAddress = '0x0001A500A6B18995B03f44bb040A5fFc28E45CB0';
+const daoAddress = '0x3C1fF68f5aa342D296d4DEe4Bb1cACCA912D95fE';
+const buOlasAddress = '0xb09CcF0Dbf0C178806Aaee28956c74bd66d21f73';
+const veOlasAddress = '0x7e01A500805f8A52Fad229b3015AD130A332B7b3';
+const valoryAddress = '0x87cc0d34f6111c8A7A4Bdf758a9a715A3675f941';
+const providerUrl = 'https://ethereum.publicnode.com';
+
+const LABELS = [
+  'veOLAS (vote escrow)',
+  'DAO Treasury',
+  'buOLAS',
+  'Valory (core contributor)',
+  'Circulating supply',
+];
+
+const ADDRESSES = [
+  veOlasAddress,
+  daoAddress,
+  buOlasAddress,
+  valoryAddress,
+];
```

Comment:
should merge these arrays into an array of objects so it's clear they're connected when they're matched by index later